src/S3Control/EndpointArnMiddleware.php

```diff
@@ -198,12 +206,16 @@ public function __invoke(CommandInterface $cmd, RequestInterface $req)
                     }
 
                     // Set modified request
-                    $req = $req
-                        ->withUri($req->getUri()->withHost($host)->withPath($path))
-                        ->withHeader('x-amz-account-id', $arn->getAccountId());
                     if (isset($body)) {
                         $req = $req->withBody($body);
                     }
+                    if ($this->isUseEndpointV2) {
```

Comment:
I feel like this logic could be reconstructed differently; maybe put all of the endpointsV2 logic into another method and just have one method that you return the result of:

if ($this->isUseEndpointV2) {
      return EndpointV2LogicMethod();
}

function EndpointV2LogicMethod() {
return nextHandler(command,request)
}